### DISCUSSION TITLE
Why is there a photo sensitive diode in the midi schematic?

### DISCUSSION THREAD

**Discussion by danielkinahan** [marcel-licence/esp32_basic_synth#104] [General]
I am looking for how to use a 6n138 with only 3.3v and I found this schematic confusing. Is the other missing diode the one that connects pins 2 and 3? What value do you suggest for the RX pull up resistor?

**Comment by marcel-licence** [comment]
The diode connected to pin 2 and 3 is for circuit protection. It would work without it but your circuit wouldn't be protected (ref. break voltage of diode).
You can try to use 1k or 10k. I think that should work.
The diode connected to pin 7 should be a light emitting diode. It would turn on when the opto-coupler gets a signal.
Hope this helps

**Reply by danielkinahan** [reply]
This does help thank you! I didn't realize you had created a schematic in your ML_Synth repo [here](https://github.com/marcel-licence/ML_SynthTools/blob/main/extras/MIDI_IN_6N138.png). For anyone else who might come across this conversation I also found a good blog post about using 6n138 with 3v3 microcontrollers [here](https://diyelectromusic.wordpress.com/2021/02/15/midi-in-for-3-3v-microcontrollers/). Although it might work sometimes with a 3v3 power supply it seems its better to just use an H11L1 or have a 5v supply.
